src/txrequest.cpp

```diff
@@ -675,6 +675,19 @@ class TxRequestTracker::Impl {
         if (it != m_index.get<ByPeer>().end()) MakeCompleted(m_index.project<ByTxHash>(it));
     }
 
+    bool ExpectedTx(NodeId peer, const uint256& txhash)
+    {
+        auto it_hash = m_index.get<ByTxHash>().lower_bound(ByTxHashView{txhash, State::REQUESTED, 0});
+        // We need to traverse all the REQUEST / COMPLETED announcements to verify we effectively
```

Comment:
`ReceivedResponse` already does this, could just have it return true in that case, and false otherwise (instead of void), and issue the error if neither the txid or wtxid resulted in a true result?

Reply:
The `ExpectedTx` have been updated to look by peer {peer, false, txhash} rather than iterating as we’re asking information for a given peer. I’ve not changed `ReceiveResponse` function call type, to avoid having unused value compiler errors, neither benchmarked yet to compare more the 2 approach.